dvc/repo/diff.py

```diff
@@ -201,18 +201,19 @@ def _calculate_renamed(new, old, added, deleted):
             bucket.append(path)
 
     renamed = []
-    for path in added:
+    for path in sorted(added, key=len):
```

Comment:
Sorting it to get the best results (in case of duplicated results). 

Before:
```console
$ dvc diff
Renamed:                                                              
    data/foo -> foo1
    foo -> data/foo1
```

After:
```console
$ dvc diff
Renamed:                                                              
    data/foo -> data/foo1
    foo -> foo1
```

Reply:
Rolled back, we should figure out a better way to associate them in case of duplicated results (trie based approach could be useful). But, I don't think it's worth the effort.